### PR TITLE
Fabric network

### PR DESCRIPTION
**Pull request by RohanShrothrium**
Implemented module for FileSystemWallet and Identity

### CODE REVIEW

**Comment by dexhunter** [comment]
@RohanShrothrium Could you write a test unit and let the ci pass?

**Comment by RohanShrothrium** [comment]
@dexhunter I've added unit tests for the module.

**Comment by dexhunter** [comment]
@RohanShrothrium Hi, could you run `make check` locally and solve the errors? Seems to be [flake8 error](https://dev.azure.com/Hyperledger/Fabric/_build/results?buildId=9297&view=logs&j=6b902995-b73d-5f5c-66fd-a7f66c857d2c&t=3ee677a2-3e85-5277-4e1a-1682471b1db5&l=1246) just make sure the code follows the flake8 guidelines

File: hfc/fabric_network/wallet.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    """
+    def __init__(self, enrollment_id, user):
+
+        if not isinstance(user, Enrollment):
+            raise ValueError('"user" is not a valid Enrollment object')
+
+        self._enrollment_id = enrollment_id
+        self._EnrollmentCert = user.cert
+        self._PrivateKey = user.private_key.private_bytes(encoding=serialization.Encoding.PEM,
+                            format=serialization.PrivateFormat.PKCS8,
```
Comment (yeasy) [inline hfc/fabric_network/wallet.py]: Guess here blocks the flake8.

**Comment by RohanShrothrium** [comment]
Resolved all flake issues.

File: hfc/fabric_network/wallet.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        """ Saves the particular Identity in the wallet """
+        sub_directory = Wallet._path + '/' + self._enrollment_id + '/'
+        os.makedirs(sub_directory, exist_ok=True)
+
+        f = open(sub_directory+'private_sk', 'wb')
+        f.write(self._PrivateKey)
+        f.close()
+
+        f = open(sub_directory+'enrollmentCert.pem', 'wb')
+        f.write(self._PrivateKey)
```
Comment (bestbeforetoday) [inline hfc/fabric_network/wallet.py]: Is it intentional that you save the private key as the certificate?

**Comment by bestbeforetoday** [comment]
It looks to me like you are implementing the Wallet from v1.4 of the Node and Java SDKs. Just be aware that the Wallet API and persistent data format are slightly different in v2.0 of those SDKs.

File: hfc/fabric_network/wallet.py (hunk 2 above)
Comment (RohanShrothrium) [reply hfc/fabric_network/wallet.py]: This was done intentionally as this is one of the few ways I found to write a Elliptic curve object!

**Comment by RohanShrothrium** [comment]
> It looks to me like you are implementing the Wallet from v1.4 of the Node and Java SDKs. Just be aware that the Wallet API and persistent data format are slightly different in v2.0 of those SDKs.

Are you suggesting to store the type of certificate also, ie. "X509" in the identity to make it into a persistent data format?

**Comment by RohanShrothrium** [comment]
> It looks to me like you are implementing the Wallet from v1.4 of the Node and Java SDKs. Just be aware that the Wallet API and persistent data format are slightly different in v2.0 of those SDKs.
According to the documentation of 2.0 wallet, the wallet stores the "X509" identities of the user. 
https://hyperledger-fabric.readthedocs.io/en/release-2.0/developapps/application.html#wallet

**Pull request merged**